dialogue-core/src/main/java/com/palantir/dialogue/core/BalancedNodeSelectionStrategyChannel.java

```diff
@@ -189,48 +291,52 @@ public void onFailure(Throwable throwable) {
             return maybe;
         }
 
-        SortableChannel computeScore() {
-            int requestsInflight = inflight.get();
-            double failureReservoir = recentFailuresReservoir.get();
-
-            // it's important that scores are integers because if we kept the full double precision, then a single 4xx
-            // would end up influencing host selection long beyond its intended lifespan in the absence of other data.
-            int score = requestsInflight + Ints.saturatedCast(Math.round(failureReservoir));
-
-            observability.debugLogComputedScore(requestsInflight, failureReservoir, score);
-            return new SortableChannel(score, this);
+        ChannelStats getSnapshot() {
+            return new ChannelStats(this, inflight.get(), recentFailuresReservoir.get(), rtt.getNanos());
         }
 
         @Override
         public String toString() {
-            return "MutableChannelWithStats{score=" + computeScore().score
+            return "MutableChannelWithStats{"
+                    + "delegate=" + delegate
                     + ", inflight=" + inflight
                     + ", recentFailures=" + recentFailuresReservoir
-                    + ", delegate=" + delegate
+                    + ", rtt=" + rtt
                     + '}';
         }
     }
 
     /**
-     * A dedicated immutable class ensures safe sorting, as otherwise there's a risk that the inflight AtomicInteger
+     * A dedicated value class ensures safe sorting, as otherwise there's a risk that the inflight AtomicInteger
      * might change mid-sort, leading to undefined behaviour.
      */
-    private static final class SortableChannel {
-        private final int score;
+    @VisibleForTesting
+    static final class ChannelStats {
```

Comment:
I think having 7 inner classes is bit too many, I think it might improve legibility if we pulled some of these out

Reply:
Yup agreed. commit incoming